assembly/src/ast/procedure/name.rs

```diff
@@ -328,3 +333,47 @@ impl FromStr for ProcedureName {
         Ok(Self(Ident::new_unchecked(Span::unknown(raw))))
     }
 }
+
+impl Serializable for ProcedureName {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        self.as_str().write_into(target)
+    }
+}
+
+impl Deserializable for ProcedureName {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        let str: String = source.read()?;
+        let proc_name =
+            ProcedureName::new_unchecked(Ident::new_unchecked(Span::unknown(Arc::from(str))));
```

Comment:
This would allow deserialization of arbitrary strings as procedure names, right? (e.g., strings with control or unprintable characters etc.) Would this not be a problem?

Should we enforce at least some kind of validation (e.g., maybe against the alphabet used on line 371 below)?

Reply:
Good point! Since it's an unchecked user input, I suppose there is a chance to use it for some sort of attack. I cannot come up with any, but I'd feel safer if we have at least some validation here. This should be a different (more relaxed) validation from the one we use when parsing the MASM source code. @bitwalker What do you think?

Reply:
I relaxed the validation rules in the `Ident` (used in `LibraryPath`) and `ProcedureName` in `new` constructors and used them in the deserialization instead of the `new_unchecked` ones.
The validation on MASM source code parsing is implemented in the lexer and does not use the checked constructors (it calls `new_unchecked`). So the existing checked constructors __are__ the different (from lexer's) kind of validation to be used everywhere else.